helm/templates/deployment.yaml

```diff
@@ -0,0 +1,62 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: coder
+  labels:
+    {{- include "coder.labels" . | nindent 4 }}
+spec:
+  # NOTE: this is currently not used as coder v2 does not support high
+  #       availability yet.
+  # replicas: {{ .Values.coder.replicaCount }}
+  replicas: 1
+  selector:
+    matchLabels:
+      {{- include "coder.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "coder.selectorLabels" . | nindent 8 }}
+    spec:
+      restartPolicy: Always
+      terminationGracePeriodSeconds: 60
+      containers:
+        - name: coder
+          image: "{{ .Values.coder.image.repo }}:{{ .Values.coder.image.tag | default (printf "v%v" .Chart.AppVersion) }}"
+          imagePullPolicy: {{ .Values.coder.image.pullPolicy }}
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          env:
+            {{- if .Values.coder.tls.secretName }}
+            - name: CODER_ADDRESS
+              value: "0.0.0.0:8443"
+            - name: CODER_TLS_ENABLE
+              value: "true"
+            - name: CODER_TLS_CERT_FILE
+              value: /etc/ssl/certs/coder/tls.crt
+            - name: CODER_TLS_KEY_FILE
+              value: /etc/ssl/certs/coder/tls.key
+            {{- else }}
+            - name: CODER_ADDRESS
+              value: "0.0.0.0:8080"
+            {{- end }}
```

Comment:
Can people just configure these with environment variables using `coder.env` instead?

Reply:
Since the port is needed for the service (and I spoke to you on Slack that we shouldn't add a port config option since it probably won't be used and we can easily add it in the future), and mounting a secret is needed for the TLS certs we can't really remove these.